### DISCUSSION TITLE
Server Installation Instructions, esp for 3.8.1

### DISCUSSION THREAD

**Discussion by gene96817** [jamulussoftware/jamulus#2088] [Documentation]
Where can I find instructions for installing the new release?
My server is running 3.8.0. I am concerned about "little" changes that are needed for 3.8.1.

Are the install instructions at https://jamulus.io/wiki/Installation-for-Linux for both client and server?
If so, it would be helpful (and reassuring) to have step-by-step instructions for getting the new server installed.

If I bungle the upgrade, were can I find instructions for installing (or reinstalling) 3.8.0?

Thanks.

**Comment by ignotus666** [comment]
Yes, those are the installation instructions, it's the same for client and server (both get installed with the single installer) and AFAIK nothing has changed regarding the installation process. That said, I'm assuming you're using a Debian-based distribution.

**Reply by gene96817** [reply]
Ok... so when I follow the instructions for Manual installation.... on step 3 with" sudo apt install ./jamulus_3.8.1_ubuntu_amd64.deb"... where is the server and client executables installed? Will the installed executable that overwrite the previous version? Will both old and new versions be retained?

**Reply by ignotus666** [reply]
It overwrites the old version. I think it gets installed into usr/bin/ but not sure. It's the same executable for both client and server - what changes is whether it's launched with the -s argument or not (-s = server).

**Reply by rdica** [reply]
`dpkg -L jamulus` will list installed files.

**Reply by gene96817** [reply]
Thanks..... and oh nuts.... now I need to test my server today. I hope I don't find any surprises. Hopefully jamulus.service is completely compatible or rehearsal tomorrow is toast.

> It overwrites the old version.

This really needs to be stated in the installation procedure and the release note!
What if I was using jamulus-headless as the server executable? Does that mean there will be a file jamulus in the same folder? 


Thanks for the reply.

**Reply by ignotus666** [reply]
I'm not familiar with headless installations, sorry. I think the policy is to always maintain backwards-compatibility though, so you should be safe. If there was anything that could break I'm pretty sure there'd have been a warning in the release notes. It was probably taken as given that installing a new version would overwrite the previous one, but I suppose there's no harm in making it clear in the documentation. Maybe you could open an issue on the website repo?

**Reply by wysocki** [reply]
I'm running a standalone server in addition to Jamulus on my laptop client (which is connected to my sound equipment. I hadn't updated the server for a while so I ran the 3.8.1 install program. It said it had to delete the old 3.6 version first so I let it. But as the install proceeded, it complained that I did not have an ASIO driver installed. Duh! I'm running a server and have no need to do any playback on it. End result was nothing got installed and now I have no server running!
I would think that the install program would be able to sense this situation and install the server without requiring ASIO but it just deleted my old server. Now what do I do?

**Reply by npostavs** [reply]
> it complained that I did not have an ASIO driver installed.

I believe it gives the option of installing even if you don't have ASIO installed. Just answer "yes" to [this question](https://github.com/jamulussoftware/jamulus/blob/r3_8_1/src/res/translation/wininstaller/en.nsi#L37):

> Jamulus needs an ASIO audio driver to work, but we couldn't find one on your PC. You should install one like ASIO4ALL (More information on jamulus.io under Installation for Windows). Do you still want to continue with the installation of Jamulus first?

**Reply by wysocki** [reply]
I did get that message and I replied YES. The install seemed to proceed ok and then I got a FINISH button along with a checkbox to run Jamulus now. When I clicked the button I got another popup saying "No ASIO audio device driver found. Please install an ASIO driver before running Jamulus...." and a QUIT button.

So I clicked the QUIT button and went into file explorer to the c:\program files (x86)\jamulus folder (where I had been running the previous version). But there was no executable there! I finally figured out that the new install went into the 64 bit program files folder, and I did find the exe there. I made corrections in the old shortcut on my desktop to point to the new installation location and it works. My fault for not noting that it was now a 64 bit installation.

But I think there is a problem in the install procedure since I'm running ONLY the server on this machine, the install program assumes I want to run the client and thus crashes with the no ASIO message when it runs jamulus at the end of the installation . I understand the same exe is used for client and server, but the install program should ask "install as client, server or both?" when it starts, and then skip over any ASIO messages if no client is selected and then run only the selected modes on exit instead of just the client when it finishes.

**Reply by henkdegroot** [reply]
Perhaps the install process is able to detect that you do not have an ASIO driver installed and then should not offer the option to start Jamulus at the end of the installation process.
However, the Jamulus Server shortcut should still have been created in the windows start menu. A shortcut on the desktop is only created for running the client, so if you had modified your desktop shortcut to launch the exe with the server option it makes sense that is no longer works.

**Reply by gilgongo** [reply]
> But I think there is a problem in the install procedure since I'm running ONLY the server on this machine, 

@wysocki I think your use case (only a server on a Windows machine) would be not have been tested for when we put the installer together. But do you mean you can't simply ignore the messages relating to the client and just (as @henkdegroot says) run the server? In which case, perhaps we need to modify the installer so that running the client at the end is an explicit step and not automatically (if I understand correctly?)

**Reply by henkdegroot** [reply]
@wysocki, I have installed on a clean windows 10 machine which does not have an asio driver installed. All seems to be working as expected.
Same as with you, I got a message indicating that Jamulus can't start and as it needs an ASIO driver.
It would make sense that we update the installation package for windows and disable running the client at the end of the install process if we detected there was no proper audio driver installed. As it is just a tick box option in the last screen, you can still manually enabled it if you would really want to. This will have the least impact on "normal" installations.

@gilgongo, what do you think about this change? I am happy to work on this (i.e. make an issue and PR).

**Reply by gilgongo** [reply]
@henkdegroot Yes, seems like a sensible thing to do. I would imagine only very small numbers of people would be affected by this issue, but it's probably worth doing. Wonder what happens on other platforms in this scenario too?

**Reply by henkdegroot** [reply]
Not sure if other platforms have the same option that is automatically starts the client after installation.

**Comment by gilgongo** [comment]
@gene96817 Seems to be some confusion here in the replies.

> Are the install instructions at https://jamulus.io/wiki/Installation-for-Linux for both client and server?

The answer to that is "no." That's for clients. BTW the reason we don't explicitly say "client installation" is we assume normal people don't know what "client" means (it's just "Jamulus" to them) and have no interest in setting up a server. 

The **correct** answer to your question is that anything to do with servers is in the [Server Administration Manual](https://jamulus.io/wiki/Running-a-Server)

See the table of contents there, but the section you need to refer to is here https://jamulus.io/wiki/Server-Linux

> If I bungle the upgrade, were can I find instructions for installing (or reinstalling) 3.8.0?

There are no instructions for that in the docs, since we assume a certain level of knowledge in wrangling Linux systems, but the short answer is to run `sudo apt-get install <package-name>=<package-version-number>` and restore the apt backup of your systemd unit file (which the 3.8.1 installer would have re-named if it had changed I think) - or use your own backup of the unit file of course.

**Reply by gene96817** [reply]
Thank you.!!!!  I will be working on the server tomorrow.
(It is so much harder when I only occasionally work on these details.)
